File: C#/Binary Tree - DFS/872. Leaf-Similar Trees.cs
/* 872. Leaf-Similar Trees
Difficulty: Easy
Submission Link: https://leetcode.com/problems/leaf-similar-trees/submissions/1141613812/ */

public class Solution {
    public void Inorder(TreeNode root, List<int> v) {
        if (root != null) {
            Inorder(root.left, v);
            if (root.left == null && root.right == null) {
                v.Add(root.val);
            }
            Inorder(root.right, v);
        }
    }

    public bool LeafSimilar(TreeNode root1, TreeNode root2) {
        List<int> v1 = new List<int>();
        List<int> v2 = new List<int>();
        Inorder(root1, v1);
        Inorder(root2, v2);
        return Enumerable.SequenceEqual(v1, v2);
    }
}

File: Python/Binary Tree - DFS/872. Leaf-Similar Trees.py
""" 872. Leaf-Similar Trees
Difficulty: Easy
Submission Link: https://leetcode.com/problems/leaf-similar-trees/submissions/1141614160/ """

class Solution(object):
    def leafSimilar(self, root1, root2):
        def inorder(root, v):
            if root:
                inorder(root.left, v)
                if not root.left and not root.right:
                    v.append(root.val)
                inorder(root.right, v)
        
        v1, v2 = [], []
        inorder(root1, v1)
        inorder(root2, v2)
        return v1 == v2